### ISSUE TITLE
ScatterPlot: Legend for continuous colorcoding

### ISSUE THREAD

**Issue by dionissqq** [datagrok-ai/public#1344]

**Comment by onuf** [comment]
@dionissqq I think this one is done (part of `v1.11.1`)? If so, please close the issue.

**Comment by dionissqq** [comment]
Done in v1.11.1

**Comment by onuf** [comment]
I think we also need to adjust the scale in the legend according to the defined min/max on the column.
![image](https://user-images.githubusercontent.com/56073962/208669244-6ec24bfe-2bc8-40d6-9f7b-6f07517356cc.png)

**Comment by onuf** [comment]
Another issue happens when applying a layout with such scatter plot - there's an error on drawing legend.
```
Invalid argument: 0
	packages/$sdk/lib/_internal/js_runtime/lib/js_helper.dart 1772:17      Object.wrapException
	packages/d4/src/color.dart 457:17                                      Object.Color.scaleColor
	packages/d4/src/scatterplot/scatterplot_core.dart 1013:13              ScatterPlotCore.getColor
	packages/d4/src/scatterplot/scatterplot_renderer.dart 135:25           _ScatterPlotCanvasRenderer.renderMarker
	packages/d4/src/scatterplot/scatterplot_core.dart 618:9                ScatterPlotCore._drawOverlay
	packages/d4/src/scatterplot/scatterplot_core.dart 517:41               ScatterPlotCore._invalidateOverlay.<fn>
```

**Comment by onuf** [comment]
Some enhancements discussed today:

- [ ] When inheriting the color-coding from the grid/df, adjustments to min/max made in the plot should be reflected in the grid. However, if the original column is not color-coded, and it's selected for the color column in the scatter plot, do not apply the configuration to the column in the grid.
- [ ] Visual improvements: the legend should occupy less space and should be moved above the column selector (more integrated into the plot itself, not as a stand-alone legend component)
- [ ] Visual improvements: show the exact min/max of the column on the axes ticks

**Comment by dnillovna** [comment]
@dionissqq were the enhancements also implemented?

**Comment by dionissqq** [comment]
enhancements also implemented in v 1.11.1. Some more visual changes might follow

**Comment by dnillovna** [comment]
Changes will be included in the release 1.12.0

**Comment by dnillovna** [comment]
@dionissqq, any updates on the ticket? Is it already done? Could you close it?

**Comment by katerina-stepanova** [comment]
v.1.11.2
Legend for continuous colour coding has some issues:
1. If a column with categorical coloring on it is used for coloring a scatter plot, both linear and categorical legends are shown:
![изображение](https://user-images.githubusercontent.com/15780643/210808970-2c78cfe8-115f-4f17-a13f-c84498ce2ed0.png)
2. Linear legend cannot be switched off by setting Legend visibility: never
![изображение](https://user-images.githubusercontent.com/15780643/210809698-5c418ce4-c088-4458-b759-1ef2e8138305.png)

**Comment by alex-aprm** [comment]
Small clarification, I've accidentally picked it from 1.12.0 to 1.11.2.
Already reverted, so think of 1.12.0

**Comment by dionissqq** [comment]
Fixed in 1.12.0
